### PR TITLE
Ensure messagebus is connected before reporting that the service is ready

### PR DESCRIPTION
**Pull request by krisgesling**
## Description
- Add standard method for starting the bus client as a daemon
- Ensure messagebus is connected before reporting that the service is ready
- Ensure exceptions during `.run()` are raised to the Enclosure service
- Add missing `.stop()` method to base Enclosure Class and shift the call of it to after exit signal
- Remove info level log presumably left over from development

To add:
- Ensure GUI messagebus is connected before reporting ready.

## How to test
Start Mycroft and check that the bus is connected before the service reports ready.

## Contributor license agreement signed?
- [x] CLA (Whether you have signed a [CLA - Contributor Licensing Agreement](https://mycroft.ai/cla/)

### CODE REVIEW

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2758))

File: mycroft/client/enclosure/base.py [hunk 1]
```diff
@@ -128,7 +131,6 @@ def handle_gui_status_request(self, message):
 
     def send(self, msg_dict):
         """ Send to all registered GUIs. """
-        LOG.info('SENDING...')
```
Comment (ken-mycroft) [inline mycroft/client/enclosure/base.py:131]: Thank god!

**Review by ken-mycroft — APPROVED** [review]

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/core/PR-2758)). 
Mycroft logs are also available: [skills.log](https://reports.mycroft.ai/core/PR-2758/logs/skills.log), [audio.log](https://reports.mycroft.ai/core/PR-2758/logs/audio.log), [voice.log](https://reports.mycroft.ai/core/PR-2758/logs/voice.log), [bus.log](https://reports.mycroft.ai/core/PR-2758/logs/bus.log), [enclosure.log](https://reports.mycroft.ai/core/PR-2758/logs/enclosure.log)

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Failed ([Results](https://reports.mycroft.ai/core/PR-2758)). 
Mycroft logs are also available: [skills.log](https://reports.mycroft.ai/core/PR-2758/logs/skills.log), [audio.log](https://reports.mycroft.ai/core/PR-2758/logs/audio.log), [voice.log](https://reports.mycroft.ai/core/PR-2758/logs/voice.log), [bus.log](https://reports.mycroft.ai/core/PR-2758/logs/bus.log), [enclosure.log](https://reports.mycroft.ai/core/PR-2758/logs/enclosure.log)

**Comment by devops-mycroft** [comment]
Voight Kampff Integration Test Succeeded  ([Results](https://reports.mycroft.ai/core/PR-2758))

**Pull request merged**